Packs/MicrosoftGraphServices/Integrations/MicrosoftGraphServices/MicrosoftGraphServices.py

```diff
@@ -0,0 +1,174 @@
+"""
+An integration to MS Graph Service Principals endpoint.
+https://docs.microsoft.com/en-us/graph/api/resources/serviceprincipal?view=graph-rest-1.0
+"""
+
+import urllib3
+
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+# Disable insecure warnings
+urllib3.disable_warnings()  # pylint: disable=no-member
+
+
+class Client:
+    def __init__(self, app_id: str, verify: bool, proxy: bool):
+        self.ms_client = MicrosoftClient(
+            self_deployed=True,
+            auth_id=app_id,
+            token_retrieval_url='https://login.microsoftonline.com/organizations/oauth2/v2.0/token',
+            grant_type=DEVICE_CODE,
+            base_url='https://graph.microsoft.com',
+            verify=verify,
+            proxy=proxy,
+            scope='offline_access Application.ReadWrite.All'
```

Comment:
why arent we using `Application.ReadWrite.OwnedBy` and `Application.Read.All` ?

Reply:
I've checked it, when I've changed the permissions to OwnedBy I couldn't use the API (no resource found).